File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
from itertools import product

def solution(word):
    
    words = []
    moeum = ['A', 'E', 'I', 'O', 'U']
    for i in range(1, 6):
        for j in product(moeum, repeat=i):
            words.append(''.join(list(j)))
            print(j)
    words.sort()
    return words.index(word)+1

File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
class Solution {
    public int solution(String word) {
        
        String str = "AEIOU";
        int[] x = {781, 156, 31, 6, 1};
        int index, result = word.length();
        for(int i=0; i<word.length(); i++) {
            index = str.indexOf(word.charAt(i));
            System.out.println(index);
            result += x[i] * index;
            //System.out.println(result);
        }
        
        return result;
    }
}